### DISCUSSION TITLE
How to find posts that have a particular tag assigned?

### DISCUSSION THREAD

**Discussion by adamnicol** [prisma/prisma#15633] [Q&A] [answered]
Hi, I'm new to Prisma and I'm trying to find posts by tag but I can't get it to work. This is my schema:

```
model Post {
  id Int @id @default(autoincrement())
  title String
  slug String
  author  User @relation(fields: [author_id], references: [id])
  author_id Int
  content String
  tags Tag[]
  comments Comment[]
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```
```
model Tag {
  id Int @id @default(autoincrement())
  name String
  posts Post[]
}
```

This is my attempt:

``` 
async function getByTag(tag: string, limit: number, offset: number): Promise<Post[]> {
  return await db.post.findMany({
    skip: offset,
    take: limit,
    where: { tags: { some: { tag: { name: tag } } } },
    orderBy: { createdAt: "desc" },
    include: {
      author: { select: { username: true } },
      _count: { select: { comments: true } },
    },
  });
}
```

This is the error I am getting:

```
error TS2322: Type '{ tag: { name: string; }; }' is not assignable to type 'TagWhereInput'. Object literal may only specify known properties, and 'tag' does not exist in type 'TagWhereInput'
```

I tried searching and found examples doing it the same way as I am so I don't know what is wrong.

**Comment by adamnicol** [comment, marked as answer]
After restarting VSCode it showed the error where before I wasn't getting an error until runtime. I then changed it back to the very first thing I tried (which VSCode was previously giving me an error for) and it now works:

```
where: { tags: { some: { name: tag } } }
```

No idea what was going on there but it's working now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
